brave/browser-laptop
# Support .torrent files in Torrent Viewer

Behold, `.torrent` file support!

![screen shot 2017-03-17 at 5 08 08 pm](https://cloud.githubusercontent.com/assets/121766/24066975/50406334-0b34-11e7-9b9f-e4bf1e1596a3.png)

Fixes: https://github.com/brave/browser-laptop/issues/6671
Fixes: https://github.com/brave/browser-laptop/issues/7772

---


Test Plan:

- Visit https://codepen.io/ferossity/full/qaezaB/
- Click the "torrent" link for any of the given files.
- Verify the following:
  1. The URL bar shows an http link that ends in ".torrent"
  2. The torrent viewer loads
  3. Clicking "Save Torrent File" causes a .torrent file to download.
  4. Clicking "Start Download" causes the download to start.
  5. Clicking "Save Torrent File" still causes a .torrent file to download.
  6. Clicking on a video/audio file inside the torrent causes streaming to immediately start.

Also, confirm that magnet links continue to work:

- Visit https://codepen.io/ferossity/full/qaezaB/
- Click the "magnet" link for any of the given files.
- Verify the following:
  1. The URL bar shows a magnet link
  2. The torrent viewer loads
  3. Clicking "Copy Magnet Link" causes the magnet link to be copied to the clipboard.
  4. Clicking "Start Download" causes the download to start.
  5. Clicking "Copy Magnet Link" still causes the magnet link to be copied to the clipboard.
  6. Clicking on a video/audio file inside the torrent causes streaming to immediately start.
